src/xcode/ENA/ENA/Source/Scenes/DataDonation/DataDonationViewModels/SettingsDataDonationViewModel.swift

```diff
@@ -0,0 +1,222 @@
+					)
+				]
+			)
+		)
+					
+		dynamicTableViewModel.add(
+			.section(separators: .all, cells: [
+				.body(
+					text: AppStrings.DataDonation.Info.dataProcessingDetails,
+					style: DynamicCell.TextCellStyle.label,
+					accessibilityIdentifier: AccessibilityIdentifiers.ExposureSubmissionQRInfo.dataProcessingDetailInfo,
+					accessibilityTraits: UIAccessibilityTraits.link,
+					action: .pushDataDonationDetails(model: DataDonationDetailsViewModel().dynamicTableViewModel,
+								  withTitle: AppStrings.DataDonation.DetailedInfo.title,
+								  completion: nil
+					),
+					configure: { _, cell, _ in
+						cell.accessoryType = .disclosureIndicator
+						cell.selectionStyle = .default
+					}),
+				.space(height: 12)
+			])
+		)
+
+		return dynamicTableViewModel
+	}
+
+	// MARK: - Internal
+
+	@objc ///  consent given switch changes
+	func didToggleDatadonationSwitch(sender: UISwitch) {
+		save(consentGiven: sender.isOn)
+	}
+
+	///  user tapped select state
+	func didTapSelectStateButton() {
+		let selectValueViewModel = SelectValueViewModel(
+			dataDonationModel.allFederalStateNames,
+			title: AppStrings.DataDonation.ValueSelection.Title.State,
+			preselected: dataDonationModel.federalStateName
+		)
+		selectValueViewModel.$selectedValue.sink { [weak self] federalState in
+			guard self?.dataDonationModel.federalStateName != federalState else {
+				return
+			}
+			// if a new fedaral state got selected reset region as well
+			self?.dataDonationModel.federalStateName = federalState
+			self?.dataDonationModel.region = nil
+			self?.dataDonationModel.save()
+		}.store(in: &subscriptions)
+		presentSelectValueList(selectValueViewModel)
+	}
+
+	///  user tapped select region
+	func didTapSelectRegionButton() {
+		guard let federalStateName = dataDonationModel.federalStateName else {
+			Log.debug("Missing federal state to load regions", log: .ppac)
+			return
+		}
+
+		let selectValueViewModel = SelectValueViewModel(
+			dataDonationModel.allRegions(by: federalStateName),
+			title: AppStrings.DataDonation.ValueSelection.Title.Region,
+			preselected: dataDonationModel.region
+		)
+		selectValueViewModel.$selectedValue .sink { [weak self] region in
+			guard self?.dataDonationModel.region != region else {
+				return
+			}
+			self?.dataDonationModel.region = region
+			self?.dataDonationModel.save()
+		}.store(in: &subscriptions)
+
+		presentSelectValueList(selectValueViewModel)
+	}
+
+	///  user tapped select age
+	func didTapAgeButton() {
+		let selectValueViewModel = SelectValueViewModel(
+			AgeGroup.allCases.map({ $0.text }),
+			title: AppStrings.DataDonation.ValueSelection.Title.Age,
+			preselected: dataDonationModel.age
+		)
+		selectValueViewModel.$selectedValue .sink { [weak self] age in
+			guard self?.dataDonationModel.age != age else {
+				return
+			}
+			self?.dataDonationModel.age = age
+			self?.dataDonationModel.save()
+//			self?.reloadTableView.toggle()
```

Comment:
can we remove this?

Reply:
yes, we can

Reply:
![image](https://user-images.githubusercontent.com/10122052/107953342-4ef19000-6f9b-11eb-953b-0c8cf44ed89f.gif)